Server/Server.Core/Server.Core.Static/Server.cpp

```diff
@@ -145,131 +259,50 @@ void Server::onMessage(const std::shared_ptr<Connection>& client, Message& messa
-void Server::waitForClientConnection()
-{   
-    _acceptor.async_accept([this](const std::error_code error, tcp::socket socket) { acceptingClientConnection(error, socket); });
+        //_connectionsPointers.erase(std::remove(_connectionsPointers.begin(), _connectionsPointers.end(), client),
+         //                          _connectionsPointers.end());
+    }
 }
 
-void Server::acceptingClientConnection(const std::error_code& error, tcp::socket& socket)
+void Server::update(size_t maxMessages, bool wait)
 {
-    if (!error)
-    {
-        std::ostringstream outputPoint;
-        outputPoint << socket.remote_endpoint();
-
-        FileLogger::getInstance().log
-        (
-            "[SERVER] New Connection: " + outputPoint.str(),
-            LogLevel::INFO
-        );
-
-        auto newConnection = 
-            std::make_shared<Connection>(Connection::OwnerType::SERVER, _context, std::move(socket), _incomingMessagesQueue);
-
-        if (onClientConnect(newConnection))
-        {
-            _connectionsPointers.emplace_back(std::move(newConnection));
-            _connectionsPointers.back()->connectToClient(_idCounter++);
+    if (wait) _incomingMessagesQueue.wait();
 
-            FileLogger::getInstance().log
-            (
-                "[" + std::to_string(_connectionsPointers.back()->getID()) 
-                + "] Connection Approved", 
-                LogLevel::INFO
-            );
-        }
-        else
+    if (_incomingMessagesQueue.size() > _criticalQueueSize)
+    {
+        for (size_t i = 0; i < _newThreadsCount; ++i)
         {
-            FileLogger::getInstance().log
-            (
-                "[SERVER] Connection Denied", 
-                LogLevel::INFO
-            );
+            _threads.emplace_back(std::thread([this]() { _context.run(); }));
         }
     }
-    else
-    {
-        FileLogger::getInstance().log
-        (
-            "[SERVER] New Connection Error: " + error.message(), 
-            LogLevel::ERR
-        );
-    }
 
-    waitForClientConnection();
-}
+    size_t messagesCount = size_t();
 
-void Server::messageClient(std::shared_ptr<Connection> client, const Message& message)
-{
-    if (client != nullptr && client->isConnected())
-    {
-        client->send(message);
-    }
-    else
+    while (messagesCount < maxMessages && !_incomingMessagesQueue.empty())
     {
-        onClientDisconnect(client);
-
-        client.reset();
+        Message messageInfo = _incomingMessagesQueue.pop_front();
 
-        _connectionsPointers.erase(std::remove(_connectionsPointers.begin(), _connectionsPointers.end(), client),
-                                   _connectionsPointers.end());
+        onMessage(messageInfo.mRemote, messageInfo);
+        messagesCount++;
     }
 }
 
-void Server::messageAllClients(const Message& message, const std::shared_ptr<Connection>& exceptionClient)
+void Server::messageAllClients(std::shared_ptr<Connection> exceptionClient, const Message& message) const
```

Comment:
Why did you add a message type variable to the method arguments?

Reply:
> Why did you add a message type variable to the method arguments?

Reply:
Why did you delete const and & from shared_ptr? I guess there we don't change state of ownership